sources/packages/backend/libs/integrations/src/services/sfas/sfas-restriction.service.ts

```diff
@@ -45,10 +50,21 @@ export class SFASRestrictionService
         select: { id: true },
         where: { restrictionCode: "LGCY" },
       });
-      await this.repo.manager.query(this.bulkInsertStudentRestrictionsSQL, [
-        legacyRestriction.id,
-        creator.id,
-      ]);
+      await this.repo.manager.transaction(async (manager) => {
+        await manager.query(this.bulkInsertLegacyRestrictionsSQL, [
```

Comment:
Is there anything that would prevent us from executing `bulkInsertLegacyRestrictionsSQL` and `bulkInsertSFASMappedRestrictionsSQL` in parallel?

Reply:
@andrewsignori-aot Correct me if I'm wrong. I thought we need to execute `bulkInsertLegacyRestrictionsSQL` first then execute `bulkInsertSFASMappedRestrictionsSQL`. If we put them into parallel with `Promise.all([])`, there is a chance that `bulkInsertSFASMappedRestrictionsSQL` could be completed before `bulkInsertLegacyRestrictionsSQL`.